### PR TITLE
Bugfix - Upgrading HTTP error handling

### PR DESCRIPTION
**Pull request by Rican7**
## HTTP error handling

This PR updates Paulus to use the newer style of HTTP error handling introduced in chriso/klein.php#16.

This is especially important since the old style of HTTP error handling has been deprecated as of chriso/klein.php#221.

---
### Composer bits

This PR also cleans up some of the composer requirements since PHP Codesniffer was yelling at me for a false-positive PSR-2 compliance issue. When I went to upgrade it, I figured I'd clean up the composer requirements to use the newly suggested style of version constraints.

This new style of version constraints is not only suggested, its now the automatic designation as of composer/composer#3096. More info on and explanation of this new suggestion can be found here:
http://blog.doh.ms/2014/10/13/installing-composer-packages/

### CODE REVIEW

File: src/Paulus/Paulus.php [hunk 1]
```diff
@@ -354,6 +359,42 @@ function ($router, $message, $class, Exception $exception) {
...
+    {
+        $this->router->onHttpError(
+            function ($code, $router, $matched, $methods_matched, $http_exception) {
+                switch ($code) {
+                    case 404:
+                        throw EndpointNotFound::create(null, null, $http_exception);
+                        break;
+                    case 405:
+                        // Don't error out on an OPTIONS request
+                        if (!$router->request()->method('OPTIONS')) {
```
Comment (brianmuse) [inline src/Paulus/Paulus.php:379]: Oh nice call

File: src/Paulus/Exception/Http/WrongMethod.php [hunk 2]
```diff
@@ -56,4 +63,28 @@ class WrongMethod extends MethodNotAllowed implements ApiVerboseExceptionInterfa
...
+    /**
+     * Set the HTTP methods that are "allowed" or possible
+     *
+     * This helps communicate to the client what methods would be correct to use
+     * when they receive this exception as an HTTP response
+     *
+     * @param array $allowed_methods
+     * @return WrongMethod
+     */
+    public function setAllowedMethods(array $allowed_methods)
```
Comment (brianmuse) [inline src/Paulus/Exception/Http/WrongMethod.php:81]: v nice

**Comment by brianmuse** [comment]
I have nothing really to add. This looks very clean and well thought out. :+1:

**Comment by Rican7** [comment]
Thanks Brian!

File: src/Paulus/Paulus.php [hunk 3]
```diff
@@ -354,6 +359,42 @@ function ($router, $message, $class, Exception $exception) {
     }
 
     /**
+     * Setup our router's HTTP error handler
+     *
+     * This uses our router's http error callback registration method to assign
+     * a basic, default handler to execute when an HTTP error occurs
+     *
```
Comment (jstruzik) [inline src/Paulus/Paulus.php:366]: Do you want to include the `@throws` tags here?

**Comment by jstruzik** [comment]
LGTM :+1: Good work on this @Rican7!

File: src/Paulus/Paulus.php (hunk 3 above)
Comment (Rican7) [inline src/Paulus/Paulus.php:366]: I would, but the method itself doesn't actually throw the error, it just registers a callable that gets called in another area. The callback is what throws the exception.

Comment (jstruzik) [inline src/Paulus/Paulus.php:366]: Makes sense

**Comment by Rican7** [comment]
Thanks guys!

Merging....

**Pull request merged**